src/simularium/ISimulator.ts

```diff
@@ -1,60 +1,49 @@
-    // a callback to notify when TrajctoryFileInfo is ready
+    /**
+     * Callbacks to subscribe front end implementations
+     * to data events and error handling from the simulator
+     */
+
+    /** a callback to notify when TrajectoryFileInfo is ready */
     setTrajectoryFileInfoHandler(
         handler: (msg: TrajectoryFileInfo) => void
     ): void;
 
-    // a callback to notify when VisDataMessage is ready (the agent data)
+    /** a callback to notify when VisDataMessage is ready (the agent data) */
     setTrajectoryDataHandler(
         handler: (msg: VisDataMessage | ArrayBuffer) => void
     ): void;
 
-    /**
-     * Connect
-     * */
-    connectToRemoteServer(address: string): Promise<string>;
-    socketIsValid(): boolean;
-    isConnectedToRemoteServer(): boolean;
-    getIp(): string;
-    disconnect(): void;
-
-    // TODO Fold into sendUpdate
-    sendTimeStepUpdate(newTimeStep: number): void;
-    sendUpdate(data: Record<string, unknown>): void;
-
-    /**
-     * Simulation Control
-     *
-     * Simulation Run Modes:
-     *  Live : Results are sent as they are calculated
-     *  Pre-Run : All results are evaluated, then sent piecemeal
-     *  Trajectory File: No simulation run, stream a result file piecemeal
-     *
-     */
-    // these should probably be abstracted to just "start"
-    // and have a separate interface for initialization?
-    // For example, "start" = initialize, and "resume" = play
-    startRemoteSimPreRun(timeStep: number, numTimeSteps: number): void;
-    startRemoteSimLive(): void;
-    startRemoteTrajectoryPlayback(fileName: string): Promise<void>;
-    sendModelDefinition(model: string): void;
-
-    requestTrajectoryFileInfo(fileName: string): void;
-
-    pauseRemoteSim(): void;
-
-    resumeRemoteSim(): void;
-
-    abortRemoteSim(): void;
+    /** a callback to propagate errors from a simulator to it's implementing context */
+    setErrorHandler(handler: (msg: Error) => void): void;
 
+    /** general update function for client simulators
+     * todo: could also be implemented for control of live
+     * and pre-run remote simulations
+     * */
+    sendUpdate(data: Record<string, unknown>): Promise<void>;
+
+    // General simulation streaming control
+    /** prepare a simulation for streaming, async to accommodate connection steps */
+    initialize(fileName: string): Promise<void>;
+    /** stop sending frames, release resources, close connections */
+    abort(): void;
+    /** request a stream of frames to begin */
+    stream(): void;
+    /** request that streaming of frames stop */
+    pause(): void;
+    /** request a single frame of data by frame number */
     requestSingleFrame(startFrameNumber: number): void;
-
-    // what do these mean for live mode?? can go backward but not forward
-    gotoRemoteSimulationTime(time: number): void;
+    /** request a single frame of data by time stamp */
+    requestFrameByTime(time: number): void;
```

Comment:
How do you feel about renaming requestSingleFrame to just requestFrame, so that it looks more like requestFrameByTime ?  (take out the word Single)